template-manager-cmtemplate/src/main/java/com/sequenceiq/cloudbreak/cmtemplate/configproviders/hbase/HbaseCloudStorageServiceConfigProvider.java

```diff
@@ -56,11 +57,12 @@ public boolean isConfigurationNeeded(CmTemplateProcessor cmTemplateProcessor, Te
         String cdhVersion = getCdhVersion(source);
         boolean is727OrNewer = isVersionNewerOrEqualThanLimited(cdhVersion, CLOUDERAMANAGER_VERSION_7_2_7);
         String accountId = ThreadBasedUserCrnProvider.getAccountId();
-        boolean sdxHbaseCloudStorageEnabled =
-                entitlementService.sdxHbaseCloudStorageEnabled(accountId);
+        boolean sdxHbaseCloudStorageEnabled = entitlementService.sdxHbaseCloudStorageEnabled(accountId);
+        boolean razEnabled = entitlementService.razEnabled(accountId);
         return source.getFileSystemConfigurationView().isPresent()
                 && cmTemplateProcessor.isRoleTypePresentInService(getServiceType(), getRoleTypes())
-                && (!datalakeCluster || (is727OrNewer && sdxHbaseCloudStorageEnabled));
+                && (!datalakeCluster || (is727OrNewer && ((CloudPlatform.AWS.equals(source.getCloudPlatform()) && (!razEnabled)) ||
+            sdxHbaseCloudStorageEnabled)));
```

Comment:
Huh, too many parentheses, hard to understand. 😄 Perhaps simplify

```
((CloudPlatform.AWS.equals(source.getCloudPlatform()) && (!razEnabled))
```

to

```
(CloudPlatform.AWS.equals(source.getCloudPlatform() && !razEnabled)
```

Reply:
One more, unrelated q: What is the expectation in case of the following?

- DL and
- CR >= 7.2.7
- AWS and
- RAZ enabled (not only the entitlement) and
- `CDP_SDX_HBASE_CLOUD_STORAGE` entitlement present

According to the logic, `CDP_SDX_HBASE_CLOUD_STORAGE` will win and still yield a `true` result. Is that what we want?

Reply:
yes. In this way, QE can test the fix for RAZ on backup by enabling the CDP_SDX_HBASE_CLOUD_STORAGE entitlement.
This code change is to stor HBase data on S3 even when the CDP_SDX_HBASE_CLOUD_STORAGE entitlement is not enabled.

Reply:
Got it, thanks.